frontend/src/components/cluster/ClusterGroupErrorMessage.tsx

```diff
@@ -0,0 +1,63 @@
+import { InlineIcon } from '@iconify/react';
+import { Box, Typography, useTheme } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { ApiError } from '../../lib/k8s/apiProxy';
+
+export interface ClusterGroupErrorMessageProps {
+  /**
+   * A message to display when clusters fail to load resources.
+   * This is used if it is passed in, otherwise a message made from clusterErrors is used.
+   */
+  message?: string;
+  /**
+   * Either an array of errors, or keyed by cluster name, valued by the error for that cluster.
+   */
+  clusterErrors?: string[] | { [cluster: string]: string | ApiError | null };
+}
+
+/**
+ * filter out null errors
+ * @returns errors, but without any that have null values.
+ */
+function cleanNullErrors(errors: ClusterGroupErrorMessageProps['clusterErrors']) {
+  if (!errors) {
+    return {};
+  }
+  const cleanedErrors: ClusterGroupErrorMessageProps['clusterErrors'] = {};
+  Object.entries(errors).forEach(([cluster, error]) => {
+    if (error !== null) {
+      cleanedErrors[cluster] = error;
+    }
+  });
+
+  return cleanedErrors;
+}
+
+export function ClusterGroupErrorMessage({
+  clusterErrors: clusters,
+  message,
+}: ClusterGroupErrorMessageProps) {
+  const { t } = useTranslation();
+  const theme = useTheme();
+  const clusterObj = cleanNullErrors(typeof clusters === 'object' ? clusters : {});
```

Comment:
type for `clusters`(`clusterErrors`) is an object or an array, but I think in case of array the `cleanNullErrors` won't properly work, so maybe restricting the type to only an object would be enough

Reply:
Thanks. Fixed.